chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -342,8 +324,8 @@ public TwitchChat(EventManager eventManager, CredentialManager credentialManager
         log.debug("Registering the following command triggers: " + commandPrefixes.toString());
 
         // register event handler
-        eventManager.onEvent(ChannelMessageEvent.class, this::onChannelMessage);
-        eventManager.onEvent(IRCMessageEvent.class, event -> {
+        eventManager.onEvent("twitch4j-chat-command-trigger", ChannelMessageEvent.class, this::onChannelMessage);
+        eventManager.onEvent("twitch4j-chat-id-name-cache", IRCMessageEvent.class, event -> {
```

Comment:
This would only register the id-name cache function once, so it would only work for a single twitchChat instance and not on the other connections right?

Reply:
Yeah for each EventManager (that is associated with at least one TwitchChat instance), we only want these event consumers to be registered once or else the consumption would be duplicated (and this could result in firing two CommandEvents when only one is appropriate, for example). With the connection pool, multiple TwitchChat instances are sharing the same EventManager, so these unique ids are necessary. Let me know if I missed the point of your question

Reply:
This was only supposed to highlight the 2nd line.

`twitch4j-chat-id-name-cache` registers a consumer function in one connection (twitchChat instance) that writes the cache entries into a map in that specific connection - (`channelIdToChannelName` and `channelNameToChannelId `), so those maps would be empty in all other connections, except for the first one as the consumer function is only registered once.

Those maps are used in each connection individually when firing the IRCMessageEvent, so in all connections except for the first one the code that uses those map in the IRCMessageEvent wouldn't work as that will always get a empty map.

Reply:
Ahh, thanks for the clarification - that's a good point. Unfortunately, removing the uniqueness would result in each connection also caching channel id/name info for other TwitchChat instances on the same EventManager. A naive sharing of the map across instances would run into synchronization issues... Should we switch to the less-performance synchronizedMap to be shared across instances or not worry about cache duplication

Reply:
Perhaps each TwitchChat instance could have a dynamically generated UUID that is included in the IRCMessageEvent. Then, when it comes time to cache id/name, the consumer would first check if the UUID of the event and the UUID of the instance match

Reply:
I'm not worried about cache duplication in this case, the map just holds channel id & name, even with a lot of channels it will have a very small footprint - therefore i would just remove the uniqueness for the consumer.